nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/validation/SetPropertiesValidation.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ */
+
+package org.nuxeo.ecm.core.bulk.validation;
+
+import static org.nuxeo.ecm.core.bulk.BulkActionValidationHelper.invalidParameterMessage;
+import static org.nuxeo.ecm.core.bulk.BulkActionValidationHelper.unknownValueMessage;
+import static org.nuxeo.ecm.core.bulk.action.SetPropertiesAction.PARAM_DISABLE_AUDIT;
+import static org.nuxeo.ecm.core.bulk.action.SetPropertiesAction.PARAM_VERSIONING_OPTION;
+
+import java.io.Serializable;
+import java.util.stream.Stream;
+
+import org.nuxeo.ecm.core.api.VersioningOption;
+import org.nuxeo.ecm.core.bulk.BulkActionValidation;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.schema.SchemaManager;
+import org.nuxeo.ecm.core.schema.types.Field;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 10.10
+ */
+public class SetPropertiesValidation implements BulkActionValidation {
+
+    @Override
+    public void validate(BulkCommand command) throws IllegalArgumentException {
+
+        // Check for unknown parameter and unknown xpath
+        for (String param : command.getParams().keySet()) {
+            if (!PARAM_DISABLE_AUDIT.equals(param) && !PARAM_VERSIONING_OPTION.equals(param)) {
```

Comment:
Could be interesting to put the two checks above before this one and use `remove`, as it you'll avoid to have this condition here, wdyt ?

Reply:
But I'm not sure we really want to remove values from the `command.getParams()` map